### PR TITLE
feat(serializers): Added environment names to ProjectSummarySerializer.

### PR DESCRIPTION
**Pull request by lauryndbrown**
The `ProjectSummarySerializer` will now contain a list of environments the project is related to or `None` if no environments are associated with the project. This is used by `DetailedOrganizationSerializer`, `OrganizationProjectsEndpoint`, and `TeamProjectsEndpoint`. 

Note: Will need to update the api documentation so that it exposes this change.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_details.py
Comment (logicalbomb) [inline tests/sentry/api/endpoints/test_organization_details.py]: Is this because we added another test case? This seems a little bizarre... is it meant to be some sort of proxy for the amount of queries a page makes?

**Review by logicalbomb — APPROVED** [review]
Probably should wait for someone else's approval, but it LGTM :-)

**Comment by lynnagara** [comment]
Do we need to add the visibility filter?

**Comment by lynnagara** [comment]
Also should probably do something like this https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_environments.py#L28-L34

File: tests/sentry/api/endpoints/test_organization_details.py
Comment (lauryndbrown) [reply tests/sentry/api/endpoints/test_organization_details.py]: Yes it appears so. We did add one more database call. https://django.readthedocs.io/en/1.6.x/topics/testing/tools.html?highlight=assertnumqueries#django.test.TransactionTestCase.assertNumQueries

**Comment by lauryndbrown** [comment]
@lynnagara and @logicalbomb let me know what you think now.

File: src/sentry/api/serializers/models/project.py
Comment (lauryndbrown) [inline src/sentry/api/serializers/models/project.py]: Two `exclude` statements here since `is_hidden` is a `models.NullBooleanField` meaning it can accept `False` and `None` as values. So excluding `True` was the cleanest method here. And putting two conditions in the same `exclude` call ANDs the conditions rather than OR.

File: tests/sentry/api/serializers/test_project.py
Comment (lauryndbrown) [inline tests/sentry/api/serializers/test_project.py]: Added additional test to show we are not including no environment or hidden environments.

Comment (logicalbomb) [reply tests/sentry/api/serializers/test_project.py]: 🎉 Moar tests!

File: src/sentry/api/serializers/models/project.py
Comment (logicalbomb) [reply src/sentry/api/serializers/models/project.py]: This seems totally reasonable. Thx @lynnagara for the heads up.

**Review by logicalbomb — APPROVED** [review]

File: src/sentry/api/serializers/models/project.py
Comment (lynnagara) [inline src/sentry/api/serializers/models/project.py]: can we do `environments_by_project.get(item.id, [])` so this is always a list?

Comment (lauryndbrown) [reply src/sentry/api/serializers/models/project.py]: Of course. :)

**Pull request merged**
